azurerm/internal/services/compute/tests/data_source_disk_encryption_set_test.go

```diff
@@ -0,0 +1,65 @@
+package tests
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+)
+
+func TestAccDataSourceAzureRMDiskEncryptionSet_basic(t *testing.T) {
+	dataSourceName := "data.azurerm_disk_encryption_set.test"
+	ri := tf.AccRandTimeInt()
+	rs := acctest.RandString(6)
+	resourceGroup := fmt.Sprintf("acctestRG-%d", ri)
+	vaultName := fmt.Sprintf("vault%d", ri)
+	keyName := fmt.Sprintf("key-%s", rs)
+	desName := fmt.Sprintf("acctestdes-%d", ri)
+	location := acceptance.Location()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMDiskEncryptionSetDestroy,
+		Steps: []resource.TestStep{
+			// These two steps are used for setting up a valid keyVault, which enables soft-delete and purge protection.
+			{
+				Config:  testAccPrepareKeyvaultAndKey(resourceGroup, location, vaultName, keyName),
+				Destroy: false,
+				Check:   resource.ComposeTestCheckFunc(),
+			},
+			// This step is not negligible, without this step, the final step will fail on refresh complaining `Disk Encryption Set does not exist`
+			{
+				PreConfig: func() { enableSoftDeleteAndPurgeProtectionForKeyvault(resourceGroup, vaultName) },
+				Config:    testAccAzureRMDiskEncryptionSet_basic(resourceGroup, location, vaultName, keyName, desName),
+			},
```

Comment:
presumably we also need a couple of steps at the end to first delete the Disk Encryption Set, then Purge & Disable Soft Delete before deleting the Key Vault?

Reply:
No, we cannot. Both soft-delete and purge-protection are irreversible. once turning on, there is no turning back.
Soft-delete means that the key vault could be recovered after deletion, if you want to get rid of it permanently, you can purge it. Soft-deleted key vault will remain in Azure for 90 days.
Purge-protection means that you cannot purge this key vault.
In this case, the key vault can be deleted (soft deleted, recoverable) but it will persist its name for 90 days.